horace_core/sqw/@sqw/run_inspector.m

```diff
@@ -0,0 +1,110 @@
+function run_inspector(w,varargin)
+% run_inspector(w)
+%
+% Creates a new window displaying the parts of an sqw dataset that came
+% from individual runs, i.e. it is an animation of the result of a "split"
+% command.
+%
+% The animation can be run using the following keyboard shortcuts (or by
+% moving the scroll bar with the mouse):
+%
+%     Enter (Return) -- play/pause video (5 frames-per-second default).
+%     Backspace -- play/pause video 5 times slower.
+%     Right/left arrow keys -- advance/go back one frame.
+%     Page down/page up -- advance/go back 10 frames.
+%     Home/end -- go to first/last frame of video.
+%
+% Currently only works for 1d and 2d sqw data
+%
+% Optional inputs (multiple combinations allowed), with preceding keywords
+%
+% run_inspector(w,'col',[c_lo,c_hi]) - for the 2d slice case, allows you
+%       to specify the limits of the colourbar. If not specified then each plot
+%       will have different colour limits, determined by the min/max intensity.
+%
+% run_inspector(w,'ax',[x_lo,x_hi,y_lo,y_hi]) - for both 1d and 2d cases,
+%       allows you to specify the limits of the x and y axes of the plot. The
+%       default behavior is to use the axes of the original (unsplit) object
+
+%RAE 30/1/15
+
+%Do some checks on the data:
+if ~isa(w,'sqw') || w.data.pix.is_filebacked()
```

Comment:
This function is a bit unclear and just seems to be doing a simple operation using argument parsing. Would suggest rewriting it to use an argparser and validateattributes to verify these things.